core/src/main/java/com/sequenceiq/cloudbreak/job/salt/StackSaltStatusCheckerJob.java

```diff
@@ -0,0 +1,157 @@
+package com.sequenceiq.cloudbreak.job.salt;
+
+import static com.sequenceiq.cloudbreak.util.Benchmark.measure;
+
+import java.util.EnumSet;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.Status;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.crn.RegionAwareInternalCrnGenerator;
+import com.sequenceiq.cloudbreak.auth.crn.RegionAwareInternalCrnGeneratorFactory;
+import com.sequenceiq.cloudbreak.common.exception.BadRequestException;
+import com.sequenceiq.cloudbreak.dto.StackDto;
+import com.sequenceiq.cloudbreak.quartz.statuschecker.job.StatusCheckerJob;
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.RotateSaltPasswordReason;
+import com.sequenceiq.cloudbreak.service.RotateSaltPasswordService;
+import com.sequenceiq.cloudbreak.service.stack.StackDtoService;
+import com.sequenceiq.flow.core.FlowLogService;
+
+import io.opentracing.Tracer;
+
+@DisallowConcurrentExecution
+@Component
+public class StackSaltStatusCheckerJob extends StatusCheckerJob {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StackSaltStatusCheckerJob.class);
+
+    @Inject
+    private StackSaltStatusCheckerJobService jobService;
+
+    @Inject
+    private StackDtoService stackDtoService;
+
+    @Inject
+    private FlowLogService flowLogService;
+
+    @Inject
+    private RegionAwareInternalCrnGeneratorFactory regionAwareInternalCrnGeneratorFactory;
+
+    @Inject
+    private RotateSaltPasswordService rotateSaltPasswordService;
+
+    public StackSaltStatusCheckerJob(Tracer tracer) {
+        super(tracer, "Stack Salt Status Checker Job");
+    }
+
+    @Override
+    protected Object getMdcContextObject() {
+        return stackDtoService.getStackViewById(getStackId());
+    }
+
+    @Override
+    protected void executeTracedJob(JobExecutionContext context) throws JobExecutionException {
+        if (flowLogService.isOtherFlowRunning(getStackId())) {
+            LOGGER.debug("StackSaltStatusCheckerJob cannot run, because flow is running for stack: {}", getStackId());
+            return;
+        }
```

Comment:
this doesn't seem necessary anyway as `RotateSaltPasswordService#triggerRotateSaltPassword` would check this